libcudacxx/include/cuda/std/__type_traits/is_arithmetic_integral.h

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___TYPE_TRAITS_IS_ARITHMETIC_INTEGRAL_H
+#define _LIBCUDACXX___TYPE_TRAITS_IS_ARITHMETIC_INTEGRAL_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__type_traits/remove_cv.h>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+template <class _Tp>
+struct __is_arithmetic_integral_impl : public false_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<signed char> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<unsigned char> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<short> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<unsigned short> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<int> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<unsigned int> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<long> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<unsigned long> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<long long> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<unsigned long long> : public true_type
+{};
+#ifndef _LIBCUDACXX_HAS_NO_INT128
+template <>
+struct __is_arithmetic_integral_impl<__int128_t> : public true_type
+{};
+template <>
+struct __is_arithmetic_integral_impl<__uint128_t> : public true_type
+{};
+#endif
+
+template <class _Tp>
+struct _CCCL_TYPE_VISIBILITY_DEFAULT __is_arithmetic_integral
+    : public integral_constant<bool, __is_arithmetic_integral_impl<remove_cv_t<_Tp>>::value>
+{};
+
+#if !defined(_CCCL_NO_VARIABLE_TEMPLATES)
+template <class _Tp>
+_CCCL_INLINE_VAR constexpr bool __is_arithmetic_integral_v = __is_arithmetic_integral<_Tp>::value;
```

Comment:
++times_bool_should_not_have_been_included_in_is_integral

Seriously though, we only enable thisin C++17 mode, so I believe we should just specialize the variable template and not use a struct